### PR TITLE
TDR-2237 Additional metadata e2e tests

### PR DESCRIPTION
**Pull request by thanhz**

### CODE REVIEW

File: src/test/resources/features/FileChecksResults.feature [hunk 1]
```diff
@@ -84,16 +84,3 @@ Feature: File Checks results Page
...
-    And an existing judgment consignment for transferring body MOCK1
-    And an existing upload of 1 files
-    And 1 of the antivirus scans for the judgment transfer have finished
-    And 1 of the FFID scans for the judgment transfer have finished
-    And 1 of the checksum scans for the judgment transfer have finished
-    And the user is logged in on the file checks results page
-    Then the user will be on a page with the title "Results of checks"
-    When the user clicks the continue button
-    Then the user will be on a page with a panel titled "Transfer complete"
-    And the judgment transfer export will be complete
```
Comment (thanhz) [inline src/test/resources/features/FileChecksResults.feature:99]: Removing this scenario as it was added before we implemented the full judgment flow.

File: src/test/resources/features/ClosureMetadata.feature [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    When the user clicks the Save and Review button
+    Then the user will be on a page with the title "Review saved changes"
+
+  Scenario: Closure metadata is deleted by a standard user
+    Given A logged in standard user
+    And an existing standard consignment for transferring body MOCK1
+    And an existing upload of 2 files
+    And an existing completed closure form
+    And the logged in user navigates to the view metadata page for closure metadata
+    Then the user will be on a page with the caption "Closure metadata"
```
Comment (TomJKing) [inline src/test/resources/features/ClosureMetadata.feature:65]: Maybe these two step could be combined to some like: `Then the user will be on the "Closure metadata" "add or edit metadata" page`

File: src/test/resources/features/ClosureMetadata.feature [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    And the user selects "27(1)" for the FOI exemption code field
+    When the user clicks the Save and Review button
+    Then the user will be on a page with the title "Review saved changes"
+
+  Scenario: Closure metadata is deleted by a standard user
+    Given A logged in standard user
+    And an existing standard consignment for transferring body MOCK1
+    And an existing upload of 2 files
+    And an existing completed closure form
+    And the logged in user navigates to the view metadata page for closure metadata
```
Comment (TomJKing) [inline src/test/resources/features/ClosureMetadata.feature:64]: Could this step be change slightly to mention that there is existing metadata for the selected file.

File: src/test/resources/features/ClosureMetadata.feature [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    And existing metadata should contain the metadata Closure Start Date with value 01/01/2000
+    And existing metadata should contain the metadata Closure status with value Open
+    And existing metadata should contain the metadata Closure Period with value 5 years
+    And existing metadata should contain the metadata FOI exemption code with value 27(1)
+    And existing metadata should contain the metadata Is the title sensitive for the public? with value No, this title can be made public
+    And existing metadata should contain the metadata Is the description sensitive for the public? with value No, this description can be made public
+    When the user clicks the Delete metadata link
+    Then the user will be on a page with the large heading "Delete closure metadata"
+    When the user clicks the Delete and return to files button
+    Then the user will be on a page with the title "Choose a file"
```
Comment (TomJKing) [inline src/test/resources/features/ClosureMetadata.feature:79]: I think here the step needs to make clear that the use chooses the file which they previously deleted the metadata for.

File: src/test/resources/features/ClosureMetadata.feature [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    When the user who did not create the consignment is logged in on the view metadata page for closure metadata
+    Then the user who did not create the consignment will see the error message "You are not permitted to see this page"
+
+  Scenario: View closure metadata page is accessed by a judgment user
+    Given A logged in judgment user
+    And an existing judgment consignment for transferring body MOCK1
+    And an existing upload of 2 files
+    And an existing completed closure form
+    When the logged in user navigates to the view metadata page for closure metadata
+    Then the user who did not create the consignment will see the error message "You are not permitted to see this page"
```
Comment (TomJKing) [inline src/test/resources/features/ClosureMetadata.feature:103]: See comment above regarding whether this is the correct test here.

File: src/test/resources/features/DescriptiveMetadata.feature [hunk 6]
```diff
@@ -0,0 +1,93 @@
+Feature: Descriptive metadata pages
+
+  Scenario: File selection page is accessed by a standard user for descriptive metadata
```
Comment (TomJKing) [inline src/test/resources/features/DescriptiveMetadata.feature:3]: Maybe reword the scenario: `Descriptive metadata file selection page ... by a standard user`

File: src/test/resources/features/DescriptiveMetadata.feature [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    And an existing upload of 2 files
+    And the logged in user navigates to the files selection page for descriptive metadata
+    Then the user will be on a page with the caption "Descriptive metadata"
+
+  Scenario: Descriptive metadata form page is accessed by a standard user
+    Given A logged in standard user
+    And an existing standard consignment for transferring body MOCK1
+    And an existing upload of 2 files
+    And the logged in user navigates to the add metadata page for descriptive metadata
+    Then the user will be on a page with the caption "Descriptive metadata"
```
Comment (TomJKing) [inline src/test/resources/features/DescriptiveMetadata.feature:15]: Maybe these two step could be combined to some like: `Then the user will be on the "Descriptive metadata" "add or edit metadata" page`

File: src/test/resources/features/DescriptiveMetadata.feature [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    And the user will be on a page with the title "Add or edit metadata"
+
+  Scenario: The user will see an error when trying to view the descriptive metadata form page for a consignment they don't own
```
Comment (TomJKing) [inline src/test/resources/features/DescriptiveMetadata.feature:18]: Maybe reword to: `User sees an error when ...`

File: src/test/resources/features/DescriptiveMetadata.feature [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    And a user who did not create the consignment
+    When the user who did not create the consignment is logged in on the add metadata page for descriptive metadata
+    Then the user who did not create the consignment will see the error message "You are not permitted to see this page"
+
+  Scenario: Descriptive metadata form page is accessed by a judgment user
+    Given A logged in judgment user
+    And an existing judgment consignment for transferring body MOCK1
+    And an existing upload of 2 files
+    When the logged in user navigates to the add metadata page for descriptive metadata
+    Then the user who did not create the consignment will see the error message "You are not permitted to see this page"
```
Comment (TomJKing) [inline src/test/resources/features/DescriptiveMetadata.feature:31]: See previous comments about whether this is the correct test.

File: src/test/resources/features/DescriptiveMetadata.feature [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+  Scenario: Descriptive metadata form is partially completed by a standard user
+    Given A logged in standard user
+    And an existing standard consignment for transferring body MOCK1
+    And an existing upload of 2 files
+    And the logged in user navigates to the add metadata page for descriptive metadata
+    Then the user will be on a page with the caption "Descriptive metadata"
+    And the user selects "English" for the Language field
```
Comment (TomJKing) [inline src/test/resources/features/DescriptiveMetadata.feature:39]: Should this not be: `And the user de-selects "English" ...`

File: src/test/resources/features/DescriptiveMetadata.feature [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+    When the user clicks the Save and Review button
+    Then the user will see a form error message "Search for and select at least one Language"
+
+  Scenario: Descriptive metadata form is fully completed by a standard user
```
Comment (TomJKing) [inline src/test/resources/features/DescriptiveMetadata.feature:43]: Maybe reword this scenario slightly to: `Standard user completes all fields on descriptive metadata form`

File: src/test/resources/features/DescriptiveMetadata.feature [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+    Given A logged in standard user
+    And an existing standard consignment for transferring body MOCK1
+    And an existing upload of 2 files
+    And the logged in user navigates to the add metadata page for descriptive metadata
+    Then the user will be on a page with the caption "Descriptive metadata"
```
Comment (TomJKing) [inline src/test/resources/features/DescriptiveMetadata.feature:48]: See above, about combining these two steps to a single check

File: src/test/resources/features/DescriptiveMetadata.feature [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+    And the user will be on a page with the title "View existing metadata"
+    And existing metadata should contain 4 values
+    And existing metadata should contain the metadata Name with value path0
+    And existing metadata should contain the metadata Description with value test description
+    And existing metadata should contain the metadata Date of the record with value 08/03/2023
+    And existing metadata should contain the metadata Language with value English
+    When the user clicks the Delete metadata link
+    Then the user will be on a page with the large heading "Delete descriptive metadata"
+    When the user clicks the Delete and return to files button
+    Then the user will be on a page with the title "Choose a file"
```
Comment (TomJKing) [inline src/test/resources/features/DescriptiveMetadata.feature:71]: Should make it clear in the steps that its the same file the user views,  from which they deleted the metadata.

File: src/test/resources/features/DescriptiveMetadata.feature [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+    When the user who did not create the consignment is logged in on the view metadata page for descriptive metadata
+    Then the user who did not create the consignment will see the error message "You are not permitted to see this page"
+
+  Scenario: View descriptive metadata page is accessed by a judgment user
+    Given A logged in judgment user
+    And an existing judgment consignment for transferring body MOCK1
+    And an existing upload of 2 files
+    And an existing completed descriptive form
+    When the logged in user navigates to the view metadata page for descriptive metadata
+    Then the user who did not create the consignment will see the error message "You are not permitted to see this page"
```
Comment (TomJKing) [inline src/test/resources/features/DescriptiveMetadata.feature:93]: See comment above about whether this is the correct test here.

**Review by TomJKing — COMMENTED** [review]
Looks good. Just comments about rewording the scenarios/steps. Thanks

File: src/test/resources/features/ClosureMetadata.feature [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+    And a user who did not create the consignment
+    When the user who did not create the consignment is logged in on the add metadata page for closure metadata
+    Then the user who did not create the consignment will see the error message "You are not permitted to see this page"
+
+  Scenario: Closure metadata form page is accessed by a judgment user
+    Given A logged in judgment user
+    And an existing judgment consignment for transferring body MOCK1
+    And an existing upload of 2 files
+    When the logged in user navigates to the add metadata page for closure metadata
+    Then the user who did not create the consignment will see the error message "You are not permitted to see this page"
```
[the remark this answers is not in the document]
Comment (thanhz) [reply src/test/resources/features/ClosureMetadata.feature]: This particular step is misleading as all it really does is look for an error message on the page. The test is using the consignment the judgment user created. I've renamed the step definition to be more representative of what it does.

File: src/test/resources/features/ClosureMetadata.feature (hunk 4 above)
Comment (thanhz) [reply src/test/resources/features/ClosureMetadata.feature]: Still figuring this one out. Refer to the above comment.

File: src/test/resources/features/ClosureMetadata.feature (hunk 5 above)
Comment (thanhz) [reply src/test/resources/features/ClosureMetadata.feature]: Refer to my previous comment. I've renamed the step definition to be more representative of what it actually does.

File: src/test/resources/features/DescriptiveMetadata.feature (hunk 9 above)
Comment (thanhz) [reply src/test/resources/features/DescriptiveMetadata.feature]: Refer to my previous comment about renaming the step definition

File: src/test/resources/features/DescriptiveMetadata.feature (hunk 10 above)
Comment (thanhz) [reply src/test/resources/features/DescriptiveMetadata.feature]: I've modified to step definition to look for both "select|de-select" and changed the wording to "de-selects"

File: src/test/resources/features/DescriptiveMetadata.feature (hunk 13 above)
Comment (thanhz) [reply src/test/resources/features/DescriptiveMetadata.feature:68]: See comment above

File: src/test/resources/features/DescriptiveMetadata.feature (hunk 14 above)
Comment (thanhz) [reply src/test/resources/features/DescriptiveMetadata.feature]: See comment above

File: src/test/resources/features/ClosureMetadata.feature (hunk 3 above)
Comment (TomJKing) [reply src/test/resources/features/ClosureMetadata.feature:63]: Could just change the name of the scenario to something like this: `Existing closure metadata for a file is deleted by a standard user`? I think that would make it sufficiently clear.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks.
